R/scale-linewidth.R

```diff
@@ -0,0 +1,85 @@
+#' Scales for line width
+#'
+#' `scale_linewidth` scales the width of lines and polygon strokes. Due to
+#' historical reasons, it is also possible to control this with the `size`
+#' aesthetic, but using `linewidth` is encourage to clearly differentiate area
+#' aesthetics from stroke width aesthetics.
```

Comment:
I'm not sure if there are any valid usage, but maybe the document or NEWS should describe how to mimic `scale_size_area()` for linewidth?

Also, probably the document on `scale_size()`'s side should be modified as well.

Reply:
I don't think we should add that - line width scaling is one-dimensional and adding anything describing it differently will just confuse users.

I'll have a look at the `scale_size()` docs

Reply:
Hmm, what do you propose for the `scale_size()` docs - I don't think it contains anything wrong... Just a short notice about using `scale_linewidth()` for strokes?

Reply:
> I don't think we should add that - line width scaling is one-dimensional and adding anything describing it differently will just confuse users.

I see. I was thinking about those who need backward-compatibility, but such users can probably keep using `scale_size()`, so please ignore.

>  what do you propose for the `scale_size()` docs - I don't think it contains anything wrong... Just a short notice about using `scale_linewidth()` for strokes?

Yes, something like that. Since `size` can still be used for strokes after this pull request gets merged, I think we should clarify when to use which. I think `scale_size()` is one of the good place for such a notice.

Reply:
yeah, backwards compatibility would be insisting on using a bad scaling for the aesthetic so I think it's ok to not make that easy